README.md

```diff
@@ -1,10 +1,278 @@
 # CactusBot
 
-## Open Source and Community Driven
+CactusBot is a next-generation ChatBot for LiveStreams.
+Harnessing the power of open-source, and extraordinary community to shape its path
 
-Including stunning, innovative functionalities, CactusBot is an entirely
-next-generation chat bot, ultimately harnessing the power of open source
-alongside an amazing community to shape its path.
+![screenshot of the bot running in a terminal](./assets/terminal.png)
 
-### Build Status
-[![Build Status](https://travis-ci.org/CactusDev/CactusBot.svg?branch=master)](https://travis-ci.org/CactusDev/CactusBot)
+# Installation
+
+## LINUX:
+***PYTHON 3.5 OR GREATER IS REQUIRED. ANY VERSION LOWER IS NOT SUPPORTED***
+
+## Setup Sepal:
+
+Sepal is the backend service that sends packets when the database updates
+so the bot can update dynamic values, like the spam config.
+
+## PreInstall
+
+Sepal requires [NodeJS](nodejs.org) (6 or higher), [RethinkDB](rethinkdb.com), and [Redis](redis.io)
+
+
+```
+git clone https://github.com/CactusDev/Sepal
+cd Sepal
+npm i -g typescript
+npm i
+```
+
+Next, open `src/configs/default.ts` and fill out all the fields.
+
+
+## Setup CactusAPI:
+
+CactusAPI controls all the configurable elements of the bot. From commands, to quotes, to the spam config
+
+```
+git clone https://github.com/CactusDev/CactusAPI
+cd CactusAPI
+cp config-example.py config.py
+```
+
+Next, open `config.py` in your favorite editor, and adjust `OAUTH_CREDENTIALS` to contain
+services for the platforms that your instance will run on.
+
+Example:
+
+```python
+OAUTH_CREDENTIALS = {
+    "beam": {
+        "CLIENT_ID": "SuperCoolClientIDFromBeamsDevLab",
+        "CLIENT_SECRET": "SuperCoolClientSecretFromBeamsDevLab"
+    }
+}
+```
+
+Next, set `RDB_DB`, `RDB_HOST`, and `RDB_PORT` to your rethink config.
+
+Next, change `SECRET_KEY` to something secure.
+
+Example:
+
+```python
+SECRET_KEY = "bUDWSNJQ8h39tWO72dC9HA5TWorLE2"
+```
+
+Next, change `AUTH_EXPIRATION` to the amount of time that you would like an authentication
+key to last for.
+
+If we wanted it to last for 1 day, 4 hours, 25 minutes, and 7 seconds:
+
+```python
+AUTH_EXPIRATION = {
+    "days": 1,
+    "hours": 4,
+    "minutes": 25,
+    "seconds": 7
+}
+```
+
+Next, install the dependencies.
+
+```
+pip3 install -r requirements.txt
+```
+
+Next, run `python3 run.py -c` to create all the tables needed.
+
+Finally, create a user in the API with a post request that contains the following data
+
+```json
+{
+    "service": "beam",
+    "token": "username",
+    "userId": 58396, // This should be the user id from the service's api
```

Comment:
We should keep the JSON valid, both for syntax highlighting *and* confusion. (People may just try to copy the JSON directly over, and would be very confused.) Place the comment outside?

```
`userId` corresponds to the user identifier from the service's API.
```